### DISCUSSION TITLE
How to set eza alias in powershell

### DISCUSSION THREAD

**Discussion by harshv5094** [eza-community/eza#687] [Q&A, Howto, Help]
I Don't know how to set eza alias in powershell. Can someone tell me how I can set alias in pwsh

**Comment by PThorpe92** [comment]
I believe you have to create a `profile.ps1` in your home directory, if one does not already exist, and then you can simply add 
```powershell
New-Alias -Name ls -Value eza --icons etc...
```

You can run `New-Alias` right in powershell too, but it will not persist across sessions I dont believe

**Comment by andremorata** [comment]
That will not work since Set-Alias does not support parameters directly.
on your $PROFILE file, add a function to invoke eza as you want and then add the aliases you want.
for example: 

``` pwsh 
function Invoke-Eza {
  param (
      [Parameter(ValueFromRemainingArguments = $true)]
      [string[]]$Args
  )
  eza --icons=always @Args
}

Set-Alias ls Invoke-Eza
```

As mentioned, you will have to place this code into your profile if you want to persist it for later sessions.
Tested on pwsh 7.4

**Reply by mloskot** [reply]
or `New-Alias -Name ls -Value Invoke-Eza -Force -Option AllScope` instead of `Set-Alias`
